### PR TITLE
refactor: better emoji parsing

### PR DESCRIPTION
**Pull request by Digital39999**
Before:
LakAbNXI4
XlH5Hulyo
(text shifted down)

After:
bRYuVSTw3

### CODE REVIEW

File: packages/core/src/util.ts [hunk 1]
```diff
@@ -38,4 +38,22 @@ export const validateImageExtension = (url: string) => {
 		throw new DiscordComponentsError(`The url of an image for discord-image-attachment should match the regex ${IMAGE_EXTENSION}`);
 };
 
-export const getGlobalEmojiUrl = (emojiName: string): Emoji | undefined => getConfig().emojis?.[emojiName];
+export const getGlobalEmojiUrl = (emojiName: string): Emoji | undefined => {
+	const globalEmoji = getConfig().emojis?.[emojiName];
+	if (globalEmoji) return globalEmoji;
+
+	const emojiRegex = /<a?:(?<name>\w+):(?<id>\d+)>/;
```
Comment (favna) [inline packages/core/src/util.ts:45]:
```suggestion
	const emojiRegex = /(?:<(?<animated>a)?:(?<name>\w{2,32}):)?(?<id>\d{17,21})>?/;
```

As taken from https://github.com/sapphiredev/utilities/blob/d8b3b84b96f10031466955e5c272337fa95e6933/packages/discord-utilities/src/lib/regexes.ts#L42

File: packages/core/src/util.ts [hunk 2]
```diff
@@ -38,4 +38,22 @@ export const validateImageExtension = (url: string) => {
...
+	const match = emojiRegex.exec(emojiName);
+
+	if (match?.groups) {
+		const { name, id } = match.groups;
+		const isAnimated = emojiName.startsWith('<a:');
```
Comment (favna) [inline packages/core/src/util.ts:50]:
```suggestion
		const { name, id, animated } = match.groups;
		const extension = Boolean(animated) ? 'gif' : 'png';
```

File: packages/core/src/util.ts [hunk 3]
```diff
@@ -38,4 +38,22 @@ export const validateImageExtension = (url: string) => {
...
+
+		return {
+			name,
+			url: `https://cdn.discordapp.com/emojis/${id}.${isAnimated ? 'gif' : 'png'}`
```
Comment (favna) [inline packages/core/src/util.ts:54]:
```suggestion
			url: `https://cdn.discordapp.com/emojis/${id}.${extension}`
```

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/core/src/util.ts (hunk 1 above)
Comment (Digital39999) [reply packages/core/src/util.ts:45]: Agreed, forgot it's an option.

**Comment by Digital39999** [comment]
Should be all done now!

**Review by favna — APPROVED** [review]

**Pull request merged**